### PR TITLE
feat: keplr integration

### PR DESCRIPTION
**Pull request by pastaghost**

### CODE REVIEW

File: examples/sandbox/index.ts [hunk 1]
```diff
@@ -174,7 +177,20 @@ $native.on("click", async (e) => {
...
+  window["wallet"] = wallet;
+  let deviceID = "nothing";
+  try {
+    deviceID = await wallet.getDeviceID();
+    $("#keyring select").val(deviceID);
+  } catch (err) {
+    console.error(err);
+  }
+});
+
```
Comment (gomesalexandre) [inline examples/sandbox/index.ts:190]: Does this somehow conflict with Keplr and needs to be in this diff?

File: packages/hdwallet-keplr/src/keplr.ts [hunk 2]
```diff
@@ -0,0 +1,372 @@
+import { Window as KeplrWindow } from "@keplr-wallet/types";
+import { ChainReference } from "@shapeshiftoss/caip";
+import * as core from "@shapeshiftoss/hdwallet-core";
+import _ from "lodash";
```
Comment (gomesalexandre) [inline packages/hdwallet-keplr/src/keplr.ts:4]: Shake dat tree
```suggestion
import isObject from "lodash/isObject";
```

**Review by gomesalexandre — COMMENTED** [review]
A few non-blocking comments but looks good and builds fine locally.
Will accept this one after testing it out locally

File: packages/hdwallet-keplr/src/keplr.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+  interface Window extends KeplrWindow {}
+}
+
+class KeplrTransport extends core.Transport {
+  public async getDeviceID() {
+    return "keplr:0";
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public call(...args: any[]): Promise<any> {
```
[the remark this answers is not in the document]
Comment (pastaghost) [reply packages/hdwallet-keplr/src/keplr.ts:20]: 👍

File: packages/hdwallet-keplr/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "@cosmjs/stargate": "0.28.1",
+    "@shapeshiftoss/caip": "3.0.0",
+    "@shapeshiftoss/hdwallet-core": "1.21.1",
+    "@shapeshiftoss/proto-tx-builder": "0.2.1",
+    "@shapeshiftoss/types": "3.1.3",
+    "base64-js": "^1.5.1",
+    "bip32": "^2.0.4",
+    "bitcoinjs-lib": "^5.1.6",
+    "bitcoinjs-message": "^2.1.0",
+    "cosmjs-types": "^0.5.0",
```
[the remark this answers is not in the document]
Comment (pastaghost) [reply packages/hdwallet-keplr/package.json:27]: 👍

**Review by gomesalexandre — DISMISSED** [review]
LGTM, all comments about unneeded deps and typings have been tackled.
Tested this locally by linking it in web and everything works there, so from a functional point of view that's a go 🍬

File: packages/hdwallet-keplr/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  "source": "src/index.ts",
+  "types": "dist/index.d.ts",
+  "scripts": {
+    "build": "tsc --build",
+    "clean": "rm -rf dist tsconfig.tsbuildinfo",
+    "prepublishOnly": "yarn clean && yarn build"
+  },
+  "dependencies": {
+    "@cosmjs/launchpad": "^0.27.1",
+    "@cosmjs/stargate": "0.28.1",
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply packages/hdwallet-keplr/package.json:18]: plz remove

File: packages/hdwallet-keplr/src/cosmos.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  return [
+    {
+      addressNList: [0x80000000 + 44, 0x80000000 + slip44ByCoin("Atom"), 0x80000000 + msg.accountIdx, 0, 0],
+    },
+  ];
+}
+
+export async function cosmosGetAddress(provider: any): Promise<string | undefined> {
+  const offlineSigner = provider.getOfflineSigner(ChainReference.CosmosHubMainnet);
+  const cosmosAddress = (await offlineSigner?.getAccounts())?.[0].address;
```
Comment (0xdef1cafe) [inline packages/hdwallet-keplr/src/cosmos.ts:61]: might have to opt chain all the way here

File: packages/hdwallet-keplr/src/keplr.ts [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+    return true;
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return true;
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return true;
+  }
```
Comment (0xdef1cafe) [inline packages/hdwallet-keplr/src/keplr.ts:51]: ?

File: packages/hdwallet-keplr/src/keplr.ts [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public cosmosNextAccountPath(msg: core.CosmosAccountPath): core.CosmosAccountPath | undefined {
+    // TODO: What do we do here?
+    return undefined;
+  }
+
+  public async osmosisSupportsNetwork(chainId = 1): Promise<boolean> {
+    return chainId === 1;
```
Comment (0xdef1cafe) [inline packages/hdwallet-keplr/src/keplr.ts:100]: 118?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: packages/hdwallet-keplr/src/cosmos.ts (hunk 6 above)
Comment (pastaghost) [reply packages/hdwallet-keplr/src/cosmos.ts:61]: 👍

File: packages/hdwallet-keplr/src/keplr.ts (hunk 7 above)
Comment (pastaghost) [reply packages/hdwallet-keplr/src/keplr.ts:51]: 👍

File: packages/hdwallet-keplr/src/keplr.ts (hunk 8 above)
Comment (pastaghost) [reply packages/hdwallet-keplr/src/keplr.ts:100]: 👍

File: packages/hdwallet-keplr/src/cosmos.ts (hunk 6 above)
Comment (pastaghost) [reply packages/hdwallet-keplr/src/cosmos.ts]: 👍

**Review by 0xdef1cafe — APPROVED** [review]

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**